Supplementary Query Error: Align buttons to the right / Update timeout message

matyax opened the pull request:
As agreed with @radiorental , the request was to:

- Update the timeout message to this shorter version.
- Align the buttons within the alert to the right.

An additional change included in these changes is to decrease the width of the alert and center it, because otherwise it would have 100% width, looking like:

![100% width](https://user-images.githubusercontent.com/1069378/229184973-5f67a048-3627-440b-9042-5031fb056df9.png)

**Who is this feature for?**

Users who use explore with the logs volume panel open and experience errors and timeouts. The main goal for all these changes has been to do a better job, clearly communicating to the user what failed. It was reported that the warning and the cryptic message was confusing for users, thinking that the actual logs query failed.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/64627

**Special notes for your reviewer:**

How it looks after these changes:

Other errors:

![Alert](https://user-images.githubusercontent.com/1069378/229186004-f7abbc36-6643-4112-b0fc-f82f02492416.png)

Other errors expanded:

![Expanded alert](https://user-images.githubusercontent.com/1069378/229186009-3ad1e984-90f6-433d-846a-7d152a686af9.png)

Timeout in a small screen:

![Small screen](https://user-images.githubusercontent.com/1069378/229186010-39016eb6-b504-4ba6-8c1e-8a2a5290094b.png)

Timeout in a bigger screen:

![Bigger screen](https://user-images.githubusercontent.com/1069378/229186013-d2867a36-e7d8-4216-91b5-1b62c15a9a8e.png)

** How to test **
Tthe easiest way is to harcode a true here:

https://github.com/grafana/grafana/blob/dc8827a10df28bc647cae0328b50707197c8edd5/public/app/features/explore/LogsVolumePanelList.tsx#L65

```javascript
const timeoutError = isTimeoutErrorResponse(logsVolumeData) || true;
```

A more real-world scenario would be to have a Loki instance with old data, request logs from a long time interval (last 30 days) and have the logs volume panel timeout on you.

matyax committed:
Update timeout message

matyax committed:
Align buttons to the right

matyax committed:
Remove test code

matyax commented on public/app/features/explore/SupplementaryResultError.tsx:
```diff
@@ -19,27 +20,52 @@ export function SupplementaryResultError(props: Props) {
 }
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
```
These styles are necessary because the `Alert` component has `overflow: auto` set for the content. https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Alert/Alert.tsx#L168

![Overflow](https://user-images.githubusercontent.com/1069378/229186599-a34cb556-adda-4205-9d64-d2906a0d4ba7.png)

ivanahuckova reviewed (commented):
Wouldn't we want to align to the left rather than middle. Is there already a pattern where we would do this with alert?

radiorental commented:
> Wouldn't we want to align to the left rather than middle. Is there already a pattern where we would do this with alert?
I discussed with the design system folks, we agreed...

- We need a better component for this type of interaction (something has happened, do you want to proceed or dismiss)
- The current notification component wasn't design for this transient inline interaction
- Left, center & right justification all look weird in their own way.  Go with Center for now.

With that in mind, this is a SHIP IT from UX.  Thanks @matyax

radiorental reviewed (approved):
SHIP IT

matyax committed:
Update test

matyax commented:
The split view is a good catch. I'll adjust that, thanks @svennergr !

matyax committed:
Add min width to error container

matyax commented:
Updated widths to play more nicely with smaller screens, and above all, split view.

![Min width](https://user-images.githubusercontent.com/1069378/229522263-6cd940da-9bc5-4d62-a203-89f9d66d05eb.png)

For smaller viewports, Grafana starts hiding things (independent of the error message).

![Eating contents](https://user-images.githubusercontent.com/1069378/229522377-f89d5d50-675e-48ef-b181-45830213abe6.png)

Demo:

![2023-04-03 15 18 58](https://user-images.githubusercontent.com/1069378/229522394-3e975f76-6e93-4bf9-a3f1-bf0ef32645a4.gif)

What do you think?

gelicia commented:
@matyax How is best to replicate a log query timeout? Throttle internet? something else?

matyax commented:
@gelicia the easiest way would be to harcode a true here:

https://github.com/grafana/grafana/blob/dc8827a10df28bc647cae0328b50707197c8edd5/public/app/features/explore/LogsVolumePanelList.tsx#L65

```javascript
const timeoutError = isTimeoutErrorResponse(logsVolumeData) || true;
```

A more real-world scenario would be to have a Loki instance with old data, request logs from a long time interval (last 30 days) and have the logs volume panel timeout on you.

Pauric commented:
DO NOT THROTTLE INTERNET FFS!

Only joking.

I don’t know why I’m getting these emails, but they’re entertaining. Probably best to remove me from the email list.

Thanks.

> On 3 Apr 2023, at 20:49, Kristina ***@***.***> wrote:
> 
> 
> @matyax <https://github.com/matyax> How is best to replicate a log query timeout? Throttle internet? something else?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/grafana/grafana/pull/65738#issuecomment-1494885694>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACP72IYNWITJ747RZJUHQ7DW7MSU5ANCNFSM6AAAAAAWO5IPEU>.
> You are receiving this because you were mentioned.
>

gelicia commented:
@radiorental to stop getting notifications from this ticket, please click the unsubscribe button on the right side. We cannot do that for you.
<img width="420" alt="Screenshot 2023-04-03 at 4 05 21 PM" src="https://user-images.githubusercontent.com/1533128/229627571-7965acbf-9708-43e8-be92-835316e18206.png">

matyax commented:
Oops, my bad. Thank you for correcting it @gelicia . Apologies @ pauric 😅

Pauric commented:
No problem Matias,

For what it’s worth, I think the button should be aligned to the right on all screens except portrait tablet & phone, where it aligns to the left under the text.  

You have not been charged for this information.

Good luck with your project.

Pauric



> On 3 Apr 2023, at 22:14, Matias Chomicki ***@***.***> wrote:
> 
> 
> Oops, my bad. Thank you for correcting it @gelicia <https://github.com/gelicia> . Apologies @ pauric 😅
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/grafana/grafana/pull/65738#issuecomment-1494989380>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACP72I62HVAUDAX4ARVTXFTW7M4U7ANCNFSM6AAAAAAWO5IPEU>.
> You are receiving this because you were mentioned.
>

matyax commented:
Thanks everyone!
